### DISCUSSION TITLE
Launching Google Sign in/Git Hub through https://cloud.cypress.io/login

### DISCUSSION THREAD

**Discussion by Chaithra-SSI** [cypress-io/cypress#28202] [Questions and Help]
Hello Team,

I am unable to work on Google Gmail Sign in/Git Hub Sign in after launching the "https://cloud.cypress.io/login".
I need to work in Gmail and Git Hub by visiting it through the "https://cloud.cypress.io/login".

Test Script is below.

it("Visits Cypress.io and Click Login with Git Hub", () => {
    // Visit the Cypress.io website
    cy.visit("https://cloud.cypress.io/login");
    cy.wait(5000)
    cy.xpath("//body/div[@id='app']/div[1]/div[1]/div[1]/div[1]/div[1]/div[1]/div[1]/form[1]/div[2]/button[1]/span[2]").click({ force: true })
    // Locate and click the "Login with GitHub" button
   cy.origin('https://github.com/login',()=>
   {
     cy.get('#login_field').type("Your Git ID")
     cy.get('#password').type("Your Password")
     cy.get('[value="Sign in"]').click({force:true})

   })

  });

Could anyone from the team help me out?

**Comment by portyu9** [comment]
Hi @Chaithra-SSI 

The Google/GitHub buttons on `cloud.cypress.io/login` authenticate **to Cypress Cloud**. They are not a general-purpose way to establish a browser session for subsequently automating Gmail or GitHub.

If what you actually need to test is **your own application using Google/GitHub authentication**, `cy.origin()` is the supported Cypress mechanism for crossing to the identity provider:

- use a dedicated test account
- keep the UI authentication flow to a small number of tests
- cache the authenticated browser state with `cy.session()`
- prefer programmatic/API authentication for the rest of the suite

Cypress specifically does **not** recommend relying on social-provider UI authentication in CI because providers such as Google/GitHub can introduce bot detection, MFA, CAPTCHA, or account restrictions.

So your `cy.origin('https://github.com', ...)` approach is valid for a controlled authentication test, but automating Cypress Cloud itself in order to then work inside Gmail/GitHub is not the right architecture.

Please mark this answer as accepted if it helped, thank you.
